Add method modeling flow to test plan (#3010)
github/vscode-codeql: docs/test-plan.md

   - Check that the editor loads and shows methods to model.
   - Check that methods are grouped per library (e.g. `rocksdbjni@7.7.3` or `asm@6.0`)
   - Check that the "Open database" link works.
   - Check that a method can have multiple models:
      - add a couple of new models for one method
      - save and check that the modeling file (use the 'open extension pack' button to open it) shows multiple methods
      - check that a 'duplicated classification' error appears in both model editor and modeling panel when a duplicate modeling occurs

>       - check that a 'conflicting classigication' error appears when a neutral model type is paired with a source kind

Reviewer: Neutral models aren't conflicting with source kinds only, they are conflicting with models that have the same `type` as the neutral model's `kind`. What about:

```suggestion
      - check that a 'conflicting classification' error appears when a neutral model type is paired with a model of the same kind
```

      - check that clicking on the error highlights the correct modeling in both the editor and the modeling panel
3. Check the Method Usage
   - Check that the Method Usage Panel opens and jumps to the correct usage when clicking on 'View' in the model editor

>    - Check that the usages are opening when clicking on a usage

Reviewer: We had a bug here where the first one would show correctly, but clicking on any other ones would still only show the first one. Would it make sense to change this to something like:

```suggestion
   - Check that the second and third usages show when clicking on them
```

## Optional Test Cases

### Modeling Flow

> 2. Check that a method can have multiple models:

Reviewer 2: The linter has spotted this needs to start with `1`, though I believe it won't actually affect the rendering it'll be nice to be consistent.
